aquacomputer_d5next.c

```diff
@@ -501,11 +505,12 @@ static int aqc_write(struct device *dev, enum hwmon_sensor_types type, u32 attr,
 		switch (attr) {
 		case hwmon_pwm_enable:
 			if (priv->fan_ctrl_offsets != NULL) {
-				if (val < 0 || val > 6)
+				if (val < 0 || val > 7)
 					return -EINVAL;
 
-				ret = aqc_set_u8_val(priv, priv->fan_ctrl_offsets[channel] - 1,
-						       val);
+				if (val > 0)
+					ret = aqc_set_u8_val(priv, priv->fan_ctrl_offsets[channel],
+						       val - 1);
```

Comment:
I think we need a comment here about _why_ we're subtracting 1.

Or something like `#define TRANSLATE_AQC_PWM_MODE(x) ((x) - 1)`